File: lab02/leds3.c
#include<stdio.h>
#include<conio.h>
#define am "16"
#define vm "20"
#define vd "21"
#define gpioam "/sys/class/gpio/gpio16/"
#define gpiovm "/sys/class/gpio/gpio20/"
#define gpiovd "/sys/class/gpio/gpio21/"
#define gpioSysfs "/sys/class/gpio/"

void writeGPIO (char filename[], char value[]){
	FILE* fp;
	fp=fopen(filename, "w+");
	fprintf(fp, "%s", value);
	fclose(fp);
}

int main (void) {
	printf (" Iniciando o programa em C para LEDs. \n");
	writeGPIO (gpioSysfs "export", am);
	usleep (100);
	writeGPIO (gpioam "direction", "out");
	writeGPIO (gpioSysfs "export", vm);
	usleep (100);
	writeGPIO (gpiovm "direction", "out");
	writeGPIO (gpioSysfs "export", vd);
	usleep (100);
	writeGPIO (gpiovd "direction", "out");

	for (int i=0, i<5, i++){
	printf ("Acendendo o LED vermelho \n");
writeGPIO (gpiovm "value", "1");
	usleep(2000);
	printf ("Desligando o LED vermelho \n");
	writeGPIO (gpiovm " value ", "0");

printf ("Acendendo o LED verde \n");
writeGPIO (gpiovd "value", "1");
	usleep(1000);
	printf ("Desligando o LED verde \n");
	writeGPIO (gpiovd " value ", "0");

	printf ("Acendendo o LED amarelo \n");
	writeGPIO (gpioam "value", "1");
	usleep(1000);
printf ("Desligando o LED amarelo \n");
writeGPIO (gpioam "value", "0");
}
	printf (" Desabilitando a gpio \n");
	writeGPIO (gpioSysfs "unexport", am);
	usleep (100);
	writeGPIO (gpioSysfs "unexport", vm);
	usleep (100);
	writeGPIO (gpioSysfs "unexport", vd);
	usleep (100);
	printf (" Fim do programa em C.\n");
}

File: leds3.py
#!/usr/bin/python3
# script baseado no c d i g o disponibilizado em:
# Derek Molloy , Exploring Raspberry Pi: Interfacing to the Real World with Embedded Linux ,
# Wiley 2016 , ISBN 978 -1 -119 -1868 -1 , http :// www. exploringrpi . com /

import sys
from time import sleep
LED_PATH="/sys/class/gpio/gpio"
SYSFS_DIR ="/sys/class/gpio/"
LED_NUMBER=""

def writeLED (filename,value,path=LED_PATH):
  fo=open(path+filename,"w")
  fo=(path+filename,"w")
  fo.write(value)
  fo.close()
  print(filename)
  return

print(" Iniciando o script Python para alterar a gpio " + LED_NUMBER + ".")

for i in ["16","20","21"]:
  print("inicio export")
  LED_NUMBER=i
  writeLED (filename="export",value=i,path=SYSFS_DIR)
  writeLED (filename ="direction",value ="out")
  print("fim export")

for j in range(1,5):
	print("Acendendo o LED vermelho")
	LED_NUMBER="20"
	writeLED(filename="value", value ="1",path=LED_PATH+LED_NUMBER+"/")
	sleep (2)
	print("Apagando o LED vermelho")
	writeLED(filename="value", value ="0",path=LED_PATH+LED_NUMBER+"/")
  
	print("Acendendo o LED verde")
	LED_NUMBER="21"
	writeLED(filename="value", value ="1",path=LED_PATH+LED_NUMBER+"/")
	sleep (1)
	print("Apagando o LED verde")
	writeLED(filename="value", value ="0",path=LED_PATH+LED_NUMBER+"/")

	print("Acendendo o LED amarelo")
	LED_NUMBER="16"
	writeLED(filename="value", value ="1",path=LED_PATH+LED_NUMBER+"/")
	sleep (1)
	print("Apagando o LED amrelo")
	writeLED(filename="value", value ="0",path=LED_PATH+LED_NUMBER+"/")
	
for i in ["16","20","21"]:
	LED_NUMBER=i
	print ("Desabilitando a gpio")
	writeLED(filename="unexport", value=LED_NUMBER, path=SYSFS_DIR)

print("Fim do script Python.")